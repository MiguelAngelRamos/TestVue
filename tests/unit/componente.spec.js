import { shallowMount } from '@vue/test-utils';
import Biblioteca from '../../src/Views/Biblioteca.vue';
import Categorias from '../../src/components/Categorias.vue';
import Libro from '../../src/components/Libro.vue';

describe('Biblioteca Componente', () => {

  it('Probando si se renderiza el componente', () => {
    // envolverlo
    const wrapper = shallowMount(Biblioteca);

    // hacemos la afirmaciÃ³n
    expect(wrapper.html()).toMatchSnapshot();
    // saca una fotografia si borramos o cambiamos algo del componente biblioteca lo va comparar con su snapshot y si encuentra
    // diferencias el test va dar error.
  });

  it('probando renderizacion de componentes hijos', () => {
    
    const wrapper = shallowMount(Biblioteca);
    const categorias = wrapper.findComponent(Categorias);
    const libro = wrapper.findComponent(Libro)

    // hacemos las afirmaciones

    expect(categorias.exists()).toBe(true);
    expect(libro.exists()).toBe(true);
  })
});