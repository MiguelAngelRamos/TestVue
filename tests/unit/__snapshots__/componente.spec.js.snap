// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Biblioteca Componente Probando si se renderiza el componente 1`] = `
<div>
  <categorias-stub></categorias-stub>
  <libro-stub></libro-stub>
</div>
`;
